mabwiser/linear.py

```diff
@@ -190,12 +192,19 @@ def _predict_contexts(self, contexts: np.ndarray, is_predict: bool,
             # Each row needs a separately seeded rng for reproducibility in parallel
             rng = create_rng(seed=seeds[index])
 
-            for arm in arms:
-                # Copy the row rng to the deep copied model in arm_to_model
-                arm_to_model[arm].rng = rng
+            # With epsilon probability set arm expectation to random value
+            if rng.rand() < self.epsilon:
+                for arm in arms:
+                    arm_to_expectation[arm] = rng.rand()
 
-                # Get the expectation of each arm from its trained model
-                arm_to_expectation[arm] = arm_to_model[arm].predict(row)
+            else:
+                # Create new seeded generator for model to ensure reproducibility
+                model_rng = create_rng(seed=seeds[index])
```

Comment:
I suggested this so let me cover. 

No we don't need both. However, note that since we now use the generator (`rng`) in line 196, it will have a different state when the model generator is set in line 204. To preserve the same output we had before and avoid fixing many tests I suggested this.

Admittedly, it would be cleaner to use generator.

Reply:
Actually ignore what I said, I'm not sure if putting l2_lambda first makes more sense.

Reply:
@bkleyn  Just to confirm my understanding: "so we don't need both and we could have used the existing rng on L196 BUT due to state if we make this change, we have to fix soo many output in tests, hence we can afford sticking to this as-is" 

Am I right?

Reply:
Yes, that's correct.